stan/math/opencl/kernel_cl.hpp

```diff
@@ -123,10 +128,14 @@ struct global_range_kernel {
    * @param args The arguments to pass to the kernel.
    * @tparam Args Parameter pack of all kernel argument types.
    */
-  auto operator()(cl::NDRange global_thread_size, Args... args) const {
+  template <typename... Margs>
```

Comment:
@seantalts I know we talked about this previously, but `Args` is `cl::Buffer, cl::Buffer, int int` etc. while `Margs` is `matrix_cl, matrix_cl, int int`. Is there some other way I should do this?

Reply:
Ah yeah, that makes sense. I think it might be worth trying to find a way to just have the one set of template params since you can derive the other from it exactly. One way to do this would be to have a TMP that acts exactly the way "maybe_get_buffer" does but on the type level.